packages/server/src/fhir/operations/agentupgrade.ts

```diff
@@ -40,15 +40,17 @@ export async function agentUpgradeHandler(req: FhirRequest): Promise<FhirRespons
 
   let timeout: number | undefined;
   if (_timeout) {
-    timeout = Number.parseInt(_timeout, 10);
+    timeout = Number.parseInt(singularize(_timeout) as string, 10);
```

Comment:
`singularlize` is "first wins" in the case of an array whereas the old query parsing logic was "last wins". Any concerns about that?

Reply:
True — I think in this case that's probably okay, though I'd also be open to changing `singularize` if we wanted to preserve the old behavior exactly